docs/fsharp-core-notes.md

```diff
@@ -1,126 +1,124 @@
 
-Each project is either an *application* or a *library*.
+This information is only applicable if you need to author something for F# running in extremely old environments. FSharp.Core is binary compatible otherwise.
 
-* Examples of application are `.exe` project or a `.dll` project that is a test project, addin, website, or an app.
+## FSharp.Core and F# scripts
 
-* Libraries are just ordinary `.dll` components (excluding those above which are applications). 
+F# scripts, executed by F# interactive, execute against the FSharp.Core deployed with the .NET SDK you are using. If you're expecting to use a more modern library feature and find that it's missing, it's likely because you have an older .NET SDK and thus an older F# Interactive. Upgrade your .NET SDK.
 
-* Scripts are not projects, just `.fsx` files, possibly referring to other files using `#load` and Libraries using `#r`
+## Guidance for package authors
 
-### Do *not* bundle FSharp.Core with a library 
+If you are authoring a NuGet package for consumption in the F# and .NET ecosystem, you already have to make a decision about functionality vs. reach by deciding what target framework(s) you support.
 
-Do _not_ include a copy of FSharp.Core with your library or package.  If you do, you will create havoc for users of
-your library.
+As an F# package author, you also need to make this decision with respect to FSharp.Core:
 
-The decision about which `FSharp.Core` a library binds to is up to the application hosting of the library.
-The library and/or library package can place constraints on this, but it doesn't decide it.
+* Targeting an earlier version of FSharp.Core increases your reach because older codebases can use it without issue
+* Targeting a newer version of FSharp.Core lets you use and extend newer features
 
-Especially, do _not_ include FSharp.Core in the ``lib`` folder of a NuGet package.
+This decision is critical, because it can have a network effect. If you choose a higher FSharp.Core version, then that also becomes a dependency for any other package that may depend on your package.
 
-### Always deploy FSharp.Core as part of a compiled application
+### Package authors should pin their FSharp.Core reference
 
-For applications, FSharp.Core is normally part of the application itself (so-called "xcopy deploy" of FSharp.Core).  
+The default templates for F# projects carry an implicit reference to FSharp.Core. This is ideal for application developers, since applications almost always want to be referencing the highest FSharp.Core available to them. As you upgrade your .NET SDK, the FSharp.Core package referenced implicitly will also be upgraded over time, since FSharp.Core is also distributed with the .NET SDK.
```

Comment:
I don't really understand why our library package templates don't pin given that's our guidance. But hey ho

Reply:
This is because the vast majority of library projects are code-holders for apps, not actual libraries meant for distribution and broad consumption.